sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -265,122 +281,142 @@ internal void StartMessageBatchReceiver(CancellationToken cancellationToken)
+                        {
+                            // Use CancellationToken.None to attempt to close the receiver even when shutting down
+                            await receiver.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+                        }
                     }
-                    catch (Exception ex)
-                    {
-                        // Log another exception
-                        _logger.LogError(ex, $"An unhandled exception occurred in the message batch receive loop");
+                }
+                catch (ObjectDisposedException)
+                {
+                    // Ignore as we are stopping the host
+                }
+                catch (TaskCanceledException)
+                    when(cancellationToken.IsCancellationRequested)
+                {
+                    // Ignore as we are stopping the host
+                    _logger.LogInformation("Message processing has been stopped or cancelled");
+                }
+                catch (Exception ex)
+                {
+                    // Log another exception
+                    _logger.LogError(ex, $"An unhandled exception occurred in the message batch receive loop");
 
-                        if (_isSessionsEnabled && receiver != null)
+                    if (_isSessionsEnabled && receiver != null)
+                    {
+                        // Attempt to close the session and release session lock to accept a new session on the next loop iteration
+                        try
                         {
-                            // Attempt to close the session and release session lock to accept a new session on the next loop iteration
-                            try
-                            {
-                                // Use CancellationToken.None to attempt to close the receiver even when shutting down
-                                await receiver.CloseAsync(CancellationToken.None).ConfigureAwait(false);
-                            }
-                            catch
-                            {
-                                // Best effort
-                                receiver = null;
-                            }
+                            // Use CancellationToken.None to attempt to close the receiver even when shutting down
+                            await receiver.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+                        }
+                        catch
+                        {
+                            // Best effort
+                            receiver = null;
                         }
                     }
                 }
-            }, cancellationToken);
+            }
+        }
+
+        private async Task ReceiveBatchAsync(ServiceBusReceiver receiver, CancellationToken cancellationToken)
+        {
+            IReadOnlyList<ServiceBusReceivedMessage> messages = await receiver.ReceiveMessagesAsync(
```

Comment:
What if this throws asyncronosly?

Reply:
Not sure I understand what you are getting at?

Reply:
Didn't we have a `try/catch` that set the completion source result to the exception in a failure case at one point?